app/src/main/java/com/kevinmost/overlauncher/app/App.java

```diff
@@ -7,9 +7,10 @@
 
 public class App extends Application {
   private static App INSTANCE;
-
+  private static AppComponent COMPONENT_INSTANCE = DaggerAppComponent.builder().appModule(INSTANCE.module).build();
```

Comment:
Can you make this an instance variable instead? No need for member variables to be static considering we use a singleton instance of App anyway. And then your `provideComponent()` can return `INSTANCE.appComponentInstance`